filestore/fsrefstore.go

```diff
@@ -34,6 +36,7 @@ type FileManager struct {
 	AllowUrls  bool
 	ds         ds.Batching
 	root       string
+	readerFact func(path string) (FileReader, error)
```

Comment:
nit: avoid shortened names, perhaps:
```suggestion
	makeReader func(path string) (FileReader, error)
```